ext/rotoscope/callsite.c

```diff
@@ -1,28 +1,43 @@
 #include "callsite.h"
 #include <ruby.h>
 #include <ruby/debug.h>
+#include <stdbool.h>
+
+static VALUE caller_frame(int *line, bool ruby_call) {
+  VALUE frames[2] = {Qnil, Qnil};
+  int lines[2] = {0, 0};
+
+  int frame_index = ruby_call ? 1 : 0;
+
+  // There is currently a bug in rb_profile_frames that
+  // causes the start argument to effectively always
+  // act as if it were 0, so we need to also get the top
+  // frame. (https://bugs.ruby-lang.org/issues/14607)
+  rb_profile_frames(0, frame_index + 1, frames, lines);
```

Comment:
For future reference: `rb_profile_frames` is currently undocumented, but appears ina a header file https://github.com/ruby/ruby/blob/6f2c516d8031e1c7f4704bd59c5349fab7f24187/include/ruby/debug.h#L26